cluster.tf

```diff
@@ -45,6 +45,19 @@ resource "aws_eks_cluster" "this" {
   ]
 }
 
+resource "aws_security_group_rule" "cluster_private_access" {
+  count       = var.create_eks && var.manage_aws_auth && var.cluster_endpoint_private_access && var.cluster_endpoint_public_access == false ? 1 : 0
+  type        = "ingress"
+  from_port   = 443
+  to_port     = 443
+  protocol    = "tcp"
+  cidr_blocks = var.cluster_endpoint_private_access_cidrs
+
+  security_group_id = aws_eks_cluster.this[0].vpc_config[0].cluster_security_group_id
+  depends_on        = [aws_eks_cluster.this[0]]
```

Comment:
Is this depends_on needed ? You already rely on `aws_eks_cluster.this[0].vpc_config[0].cluster_security_group_id` for `security_group_id` you already have implicit dependency.